brave/brave-browser
# Command to reset non-patched chromium files

![image](https://user-images.githubusercontent.com/741836/63488102-9bb9e080-c462-11e9-9d53-02c212fbbd10.png)


Shows a diff of each modified chromium file and asks if the user wants to reset that file. This makes sure only files that are modified are reset, if desired, and also gives an opportunity for the developer to not lose work unintentionally.

The main scenario this fixes and that I have used this command for:
 - Developer wants to reset all chromium files to the current brave-core patched state, but doesn't want to end up touching every patched file by performing a hard reset, which would result in a lengthy build. Right now, they could run `git reset --hard` in `src` directory (or run `npm run init` which will run that same git reset command), but that will result in a very long subsequent build time. Now, they can run both `npm run sync` and this new command, `npm run reset_non_patched`, will make sure only the few user-local-modified files (either that are patched by brave-core or others) are reset, and not the entire chromium repository.

Another scenario this fixes, though not perfectly:
- A scenario whereby a developer can be editing chromium files in an attempt to debug an issue or create a feature, but doesn't know exactly which changes they wish to keep. Instead of running `npm run update_patches` and having to decide which patches to keep and dismiss and then resetting the other changes manually, this command will allow them to make the decision about which changes to keep or discard first, before committing to create patches.

Fix https://github.com/brave/brave-browser/issues/5757

## Test Plan:
This adds a new command and does not affect any feature code. Build and sync should still be tested since it edits libs used by those scripts.

If you would like to use the new command:

1. Modify a chromium file (that is not already being patched by brave-core
2. Run `npm run reset_non_patched`
3. Observe that the script outputs a question asking if you want to reset each modified file. Entering `Y` should reset that file at the end, and `N` should not.


## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on



## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
